infrastructure/code/sast_enabled_test.go

```diff
@@ -39,119 +38,47 @@ func TestIsSastEnabled(t *testing.T) {
-	})
-
-	t.Run("should return true if Snyk Code is enabled and the API returns true", func(t *testing.T) {
+	t.Run("should return true if Snyk Code is enabled", func(t *testing.T) {
 		config.CurrentConfig().SetSnykCodeEnabled(true)
-		apiClient.ApiError = nil
-		apiClient.CodeEnabled = true
-		apiClient.LocalCodeEngineEnabled = false
-
-		enabled := scanner.isSastEnabled()
+		mockedSastResponse.SastEnabled = true
+		enabled := scanner.isSastEnabled(mockedSastResponse)
 
 		assert.True(t, enabled)
 	})
 
-	t.Run("should return false if Snyk Code is enabled and the API returns false", func(t *testing.T) {
-		config.CurrentConfig().SetSnykCodeEnabled(true)
-		apiClient.ApiError = nil
-		apiClient.CodeEnabled = false
-		apiClient.LocalCodeEngineEnabled = false
-
-		enabled := scanner.isSastEnabled()
-
-		assert.False(t, enabled)
-	})
-
-	t.Run("should return false if Snyk Code is enabled and the API returns an error", func(t *testing.T) {
-		config.CurrentConfig().SetSnykCodeEnabled(true)
-		apiClient.ApiError = &snyk_api.SnykApiError{}
-		apiClient.CodeEnabled = false
-		apiClient.LocalCodeEngineEnabled = false
-
-		enabled := scanner.isSastEnabled()
-
-		assert.False(t, enabled)
-	})
-
-	t.Run("should return false if Snyk Code is enabled and LocalCodeEngine is enabled", func(t *testing.T) {
-		config.CurrentConfig().SetSnykCodeEnabled(true)
-		apiClient.ApiError = nil
-		apiClient.CodeEnabled = true
-		apiClient.LocalCodeEngineEnabled = true
-
-		enabled := scanner.isSastEnabled()
-
-		assert.False(t, enabled)
-	})
-
-	t.Run("should send a warning notification if Snyk Code Local Engine is enabled", func(t *testing.T) {
-		apiClient.CodeEnabled = true
-		apiClient.LocalCodeEngineEnabled = true
-		apiClient.ApiError = nil
-		notifier := notification.NewNotifier()
-		// overwrite scanner, as we want our separate notifier
-		scanner := &Scanner{
-			SnykApiClient: apiClient,
-			errorReporter: error_reporting.NewTestErrorReporter(),
-			notifier:      notifier,
-		}
-		config.CurrentConfig().SetSnykCodeEnabled(true)
-		channel := make(chan any)
-		notifier.CreateListener(func(params any) {
-			channel <- params
-		})
-		defer notifier.DisposeListener()
-		expectedNotification := sglsp.ShowMessageParams{Type: sglsp.Warning, Message: localCodeEngineWarning}
-
-		scanner.isSastEnabled()
-
-		assert.Eventuallyf(
-			t,
-			func() bool { return expectedNotification == <-channel },
-			5*time.Second,
-			time.Millisecond,
-			"expected warning notification",
-		)
-	})
 
 	t.Run("should send a ShowMessageRequest notification if Snyk Code is enabled and the API returns false",
 		func(t *testing.T) {
-			apiClient.CodeEnabled = false
-			apiClient.LocalCodeEngineEnabled = false
-			apiClient.ApiError = nil
+			mockedSastResponse.SastEnabled = false
+
 			config.CurrentConfig().SetSnykCodeEnabled(true)
 			notifier := notification.NewNotifier()
 			// overwrite scanner, as we want our separate notifier
-			scanner := &Scanner{
```

Comment:
This is most likely needed, as the notifier is injected into the scanner, and you want to grab the notification in the test assertion.

Reply:
Oh yeah, this fixed the hanging test issue [here](https://snyk.slack.com/archives/C04ERK5JXL6/p1686582024097799?thread_ts=1686314119.352239&cid=C04ERK5JXL6). Thank you!